src/encoded/static/libs/ui/modal.js

```diff
@@ -343,24 +331,52 @@ export class ModalElement extends React.Component {
                 e.preventDefault();
             }
         }
-    }
+    };
 
-    render() {
-        return ReactDOM.createPortal(
-            <div>
-                <div className="modal" style={{ display: 'block' }}>
-                    <div className={`modal-dialog${this.props.addClasses ? ` ${this.props.addClasses}` : ''}`} role="alertdialog" aria-modal="true" aria-labelledby={this.props.labelId} aria-describedby={this.props.descriptionId}>
-                        <div className="modal-content">
-                            {this.props.modalChildren}
-                        </div>
+    React.useEffect(() => {
+        // Modal HTML gets injected at the end of <body> so the backdrop overlay works, even with other
+        // fixed elements on the page. The "modal-root" div is defined in app.js at the end of the
+        // <body> section.
+        const modalRoot = document.getElementById('modal-root');
+        modalRoot.appendChild(getModalEl());
+
+        // Focus screen reader on the given element if specified.
+        if (focusId) {
+            const focusElement = document.getElementById(focusId);
```

Comment:
Maybe use ref?

Reply:
You can for an external component; a little bit tricky. I’ll leave this for now since it’s existing code.